eta/core/c3d.py

```diff
@@ -0,0 +1,340 @@
+        self.sess = sess or tf.Session()
+        self.imgs = imgs or tf.placeholder(tf.float32, [None, 16, 112, 112, 3])
+        with tf.variable_scope('var_name') as var_scope:
+            weights = {
+                'wc1': self._variable_with_weight_decay('wc1', [3, 3, 3, 3, 64], 0.04, 0.00),
+                'wc2': self._variable_with_weight_decay('wc2', [3, 3, 3, 64, 128], 0.04, 0.00),
+                'wc3a': self._variable_with_weight_decay('wc3a', [3, 3, 3, 128, 256], 0.04, 0.00),
+                'wc3b': self._variable_with_weight_decay('wc3b', [3, 3, 3, 256, 256], 0.04, 0.00),
+                'wc4a': self._variable_with_weight_decay('wc4a', [3, 3, 3, 256, 512], 0.04, 0.00),
+                'wc4b': self._variable_with_weight_decay('wc4b', [3, 3, 3, 512, 512], 0.04, 0.00),
+                'wc5a': self._variable_with_weight_decay('wc5a', [3, 3, 3, 512, 512], 0.04, 0.00),
+                'wc5b': self._variable_with_weight_decay('wc5b', [3, 3, 3, 512, 512], 0.04, 0.00),
+                'wd1': self._variable_with_weight_decay('wd1', [8192, 4096], 0.04, 0.001),
+                'wd2': self._variable_with_weight_decay('wd2', [4096, 4096], 0.04, 0.002),
+                'out': self._variable_with_weight_decay('wout', [4096, 101], 0.04, 0.005)
+                }
+            biases = {
+                'bc1': self._variable_with_weight_decay('bc1', [64], 0.04, 0.0),
+                'bc2': self._variable_with_weight_decay('bc2', [128], 0.04, 0.0),
+                'bc3a': self._variable_with_weight_decay('bc3a', [256], 0.04, 0.0),
+                'bc3b': self._variable_with_weight_decay('bc3b', [256], 0.04, 0.0),
+                'bc4a': self._variable_with_weight_decay('bc4a', [512], 0.04, 0.0),
+                'bc4b': self._variable_with_weight_decay('bc4b', [512], 0.04, 0.0),
+                'bc5a': self._variable_with_weight_decay('bc5a', [512], 0.04, 0.0),
+                'bc5b': self._variable_with_weight_decay('bc5b', [512], 0.04, 0.0),
+                'bd1': self._variable_with_weight_decay('bd1', [4096], 0.04, 0.0),
+                'bd2': self._variable_with_weight_decay('bd2', [4096], 0.04, 0.0),
+                'out': self._variable_with_weight_decay('bout', [101], 0.04, 0.0),
+                }
+        self.build_c3d(self.imgs, self.config.dropout, self.config.batchsize,
+            weights, biases)
+        self._load_model()
+
+    def _variable_on_cpu(self, name, shape, initializer):
+        with tf.device('/cpu:0'):
+            var = tf.get_variable(name, shape, initializer=initializer)
+        return var
+
+    def _variable_with_weight_decay(self, name, shape, stddev, wd):
+        var = self._variable_on_cpu(name, shape, tf.truncated_normal_initializer(stddev=stddev))
+        if wd is not None:
+            weight_decay = tf.nn.l2_loss(var) * wd
+            tf.add_to_collection('losses', weight_decay)
+        return var
+
+    def conv3d(self, name, l_input, w, b):
```

Comment:
these should be private methods (`_conv3d`, `_max_pool`, `_build_c3d`)

Reply:
Addressed